addons/sale_stock/sale_stock.py

```diff
@@ -309,25 +328,9 @@ def product_id_change_with_wh(self, cr, uid, ids, pricelist, product, qty=0,
         warning_msgs = res_packing.get('warning') and res_packing['warning']['message'] or ''
 
         if product_obj.type == 'product':
+            warehouse = warehouse_obj.browse(cr, uid, warehouse_id, context=context) if warehouse_id else False
```

Comment:
"_check_routing" isn't using "warehouse_id" and is mostly using warehouse browse record, so I decided against passing warehouse_id. I will modify if you want.